holopy/core/tests/test_metadata.py

```diff
@@ -0,0 +1,666 @@
+        data1 = update_metadata(make_data(seed=1), **METADATA_VALUES)
+        data2 = update_metadata(make_data(seed=2), **METADATA_VALUES)
+        data = [data1, data2]
+
+        concatenated = clean_concat(data, 'point')
+        for key in METADATA_VALUES.keys():
+            self.assertIn(key, concatenated.attrs)
+            self.assertTrue(hasattr(concatenated, key))
+
+    @attr("fast")
+    def test_preserves_metadata_values(self):
+        data1 = update_metadata(make_data(seed=1), **METADATA_VALUES)
+        data2 = update_metadata(make_data(seed=2), **METADATA_VALUES)
+        data = [data1, data2]
+
+        concatenated = clean_concat(data, 'point')
+        for key, value in METADATA_VALUES.items():
+            if key != 'illum_polarization':
+                self.assertEqual(getattr(concatenated, key), value)
+        polarization_ok = np.all(
+            concatenated.illum_polarization[:2] ==
+            METADATA_VALUES['illum_polarization'])
+        self.assertTrue(polarization_ok)
+
+
+class TestUpdateMetadata(unittest.TestCase):
+    @attr("fast")
+    def test_does_update_medium_index(self):
+        detector = detector_grid(3, 0.1)
+
+        np.random.seed(10)
+        medium_index = 1 + np.random.rand()
+        updated_detector = update_metadata(detector, medium_index=medium_index)
+        self.assertEqual(updated_detector.medium_index, medium_index)
+
+    @attr("fast")
+    def test_does_update_illum_wavelength(self):
+        detector = detector_grid(3, 0.1)
+
+        np.random.seed(11)
+        illum_wavelen = np.random.rand()
+        updated_detector = update_metadata(
+            detector, illum_wavelen=illum_wavelen)
+        self.assertEqual(updated_detector.illum_wavelen, illum_wavelen)
+
+    @attr("fast")
+    def test_does_update_illum_polarization(self):
+        detector = detector_grid(3, 0.1)
+        np.random.seed(12)
+        illum_polarization = np.random.randn(2)
+        illum_polarization /= np.linalg.norm(illum_polarization)
+        updated_detector = update_metadata(
+            detector, illum_polarization=illum_polarization)
+        is_ok = np.all(
+            updated_detector.illum_polarization.values[:2] ==
+            illum_polarization)
+        self.assertTrue(is_ok)
+
+    @attr("fast")
+    def test_does_update_noise_sd(self):
+        detector = detector_grid(3, 0.1)
+        np.random.seed(13)
+        noise_sd = np.random.rand()
+        updated_detector = update_metadata(detector, noise_sd=noise_sd)
+        self.assertEqual(updated_detector.noise_sd, noise_sd)
+
+    @attr("fast")
+    def test_does_update_normals(self):
+        detector = detector_grid(3, 0.1)
+        np.random.seed(13)
+        normals = np.random.randn(3)
+        normals /= np.linalg.norm(normals)
+        updated_detector = update_metadata(detector, normals=normals)
+        self.assertTrue(np.all(updated_detector.normals.values == normals))
+
+    @attr("fast")
+    @unittest.skip("Fails")
```

Comment:
4. This test fails, because right now normals for each point aren't supported in ```detector_points```. Given that there are pathways where ```detector_points``` specifies a format for each point, it seems weird to me that you can't modify this down the line.

Thoughts? Tag as part of a new refactor issue?

Reply:
It would be nice to raise an error if you try to pass in normals instead.

Reply:
Should be fixed by 9e143ce